includes/express-checkout/class-wc-payments-express-checkout-ajax-handler.php

```diff
@@ -93,6 +95,57 @@ public function ajax_get_shipping_options() {
 		wp_send_json( $data );
 	}
 
+	/**
+	 * Get cart details.
+	 */
+	public function ajax_get_cart_details() {
+		check_ajax_referer( 'wcpay-get-cart-details', 'security' );
+
+		if ( ! defined( 'WOOCOMMERCE_CART' ) ) {
+			define( 'WOOCOMMERCE_CART', true );
+		}
+
+		if ( ! defined( 'WOOCOMMERCE_CHECKOUT' ) ) {
+			define( 'WOOCOMMERCE_CHECKOUT', true );
+		}
+
+		WC()->cart->calculate_totals();
+
+		wp_send_json(
+			array_merge(
+				$this->express_checkout_button_helper->build_display_items(),
+				[
+					'needs_shipping' => WC()->cart->needs_shipping(),
+				]
+			)
+		);
+	}
+
+	/**
+	 * Update shipping method.
+	 */
+	public function ajax_update_shipping_method() {
+		check_ajax_referer( 'wcpay-update-shipping-method', 'security' );
+
+		if ( ! defined( 'WOOCOMMERCE_CART' ) ) {
+			define( 'WOOCOMMERCE_CART', true );
+		}
+
+		$shipping_methods = filter_input( INPUT_POST, 'shipping_method', FILTER_DEFAULT, FILTER_REQUIRE_ARRAY );
+		$this->express_checkout_button_helper->update_shipping_method( $shipping_methods );
+
+		WC()->cart->calculate_totals();
+
+		$product_view_options      = filter_input_array( INPUT_POST, [ 'is_product_page' => FILTER_SANITIZE_SPECIAL_CHARS ] );
+		$should_show_itemized_view = ! isset( $product_view_options['is_product_page'] ) ? true : filter_var( $product_view_options['is_product_page'], FILTER_VALIDATE_BOOLEAN );
+
+		$data           = [];
+		$data          += $this->express_checkout_button_helper->build_display_items( $should_show_itemized_view );
+		$data['result'] = 'success';
+
+		wp_send_json( $data );
+	}
```

Comment:
These handlers are copied verbatim from `includes/class-wc-payments-payment-request-button-handler.php`, aside from minor tweaks to use `$this->express_checkout_button_helper` where appropriate.

I think this copy-paste is fine, but maybe we should try to attempt some sort of re-use? I'm leaning towards leaving this as is though.